### PR TITLE
Some Jenkins fixes

### LINKED ISSUE #2273: Jest test randomly fail on CI
**Describe the bug**
Every now and then the development tests fail on CI with the below error;

```
Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:

     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.

     • If you need a custom transformation specify a "transform" option in your config.

     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:

    https://jestjs.io/docs/en/configuration.html

```

**To Reproduce**
Unfortunately this bug is completely random

**Screenshots**
image

### PR DESCRIPTION
**Pull request by andrew-nowak**
Resolves: #2273

**Code changes:**

- [`xvfb-run -a`](http://manpages.ubuntu.com/manpages/xenial/man1/xvfb-run.1.html#options) will try to automatically pick an available port when running xvfb-run. This should hopefully reduce/remove the amount of `(electron:13016): Gtk-WARNING **: cannot open display: :99`. NB this was already being used in travis builds

- [`perl -pe 's/\\x1b\\[[0-9;]*m//g'`](https://superuser.com/a/380778) will strip the colour codes from the output of cypress (since it doesn't seem possible to configure that directly on cypress, this is the next best thing) and make the output much more legible in Jenkins

- `rm -rf pack` - fixes the cause of the "Jest unexpected token" error - that the pack directory continues to exist between runs

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Review by j-pendlebury — APPROVED** [review]
this looks good to me, but someone more familiar with infra stuff should probably take a look

cc @hindsc52 @dr3

**Review by chris-hinds — APPROVED** [review]
LGTM 👍

**Comment by chris-hinds** [comment]
This doesn't need testing.

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "build:test": "cp envConfig/test.env .env && npm run webpack:prod",
     "build:live": "cp envConfig/live.env .env && npm run webpack:prod",
     "cypress:proxy": "export NO_PROXY=localhost,localhost.bbc.com && cypress",
-    "cypress": "npm run cypress:proxy -- run",
+    "cypress": "npm run cypress:proxy -- run | perl -pe 's/\\x1b\\[[0-9;]*m//g'",
```
Comment (dr3) [inline package.json]: This means we dont have colour locally :( can you add this perl thing onto a make command please, so it only affects CI

**Review by dr3 — CHANGES_REQUESTED** [review]

**Review by dr3 — APPROVED** [review]

**Review by DenisHdz — APPROVED** [review]

**Comment by andrew-nowak** [comment]
:( moving the colour stripping functionality to the makefile seems to cause bbc-a11y to hang - will try again

File: package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
     "build:test": "cp envConfig/test.env .env && npm run webpack:prod",
     "build:live": "cp envConfig/live.env .env && npm run webpack:prod",
     "cypress:proxy": "export NO_PROXY=localhost,localhost.bbc.com && cypress",
-    "cypress": "npm run cypress:proxy -- run",
+    "cypress": "test -z $CI && npm run cypress:proxy -- run || npm run cypress:ci",
+    "cypress:ci": "npm run cypress:proxy -- run | perl -pe 's/\\x1b\\[[0-9;]*m//g'",
```
Comment (jamesbhobbs) [inline package.json:17]: You've added this but not used it. You mean to use it on travis and on jenkins don't you? I think using it in test:prod:ci script would be the fix.

**Review by jamesbhobbs — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (andrew-nowak) [reply package.json:17]: I'm using it in the line above :) (although, looking at that line again, it's wrong... let me fix that)

File: package.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
     "build:live": "cp envConfig/live.env .env && npm run webpack:prod",
     "cypress:proxy": "export NO_PROXY=localhost,localhost.bbc.com && cypress",
     "cypress": "npm run cypress:proxy -- run",
+    "cypress:ci": "npm run cypress:proxy -- run | perl -pe 's/\\x1b\\[[0-9;]*m//g'",
```
Comment (DemonusPC) [inline package.json:17]: what the hell is perl doing in here?

**Review by DemonusPC — APPROVED** [review]
Dank

File: package.json (hunk 3 above)
Comment (andrew-nowak) [reply package.json:17]: (Discussed offline, repeating here for visibility) Strips out all ANSI colour codes, making cypress output in jenkins logs legible

**Review by jamesbhobbs — APPROVED** [review]

**Pull request merged**